Description: do not use SERVER_PORT, it is confusing
  $ curl http://127.0.0.1:8080 -H 'X-Forwarded-Proto: https' -I -H 'Host: wiki.zalora.com'                               
  HTTP/1.1 301 Moved Permanently
  ....
  Location: https://wiki.zalora.com:80/Main_Page <-- the port is incorrect
  Content-Type: text/html; charset=utf-8

  See http://stackoverflow.com/questions/4433229/incorrect-port-number-returned-by-serverserver-port

Index: mediawiki-1.23.9/includes/WebRequest.php
===================================================================
--- mediawiki-1.23.9.orig/includes/WebRequest.php
+++ mediawiki-1.23.9/includes/WebRequest.php
@@ -182,11 +182,7 @@ class WebRequest {
 				continue;
 			}
 			$host = $parts[0];
-			if ( $parts[1] === false ) {
-				if ( isset( $_SERVER['SERVER_PORT'] ) ) {
-					$port = $_SERVER['SERVER_PORT'];
-				} // else leave it as $stdPort
-			} else {
+			if ( $parts[1] !== false ) {
 				$port = $parts[1];
 			}
 			break;
